inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -98,12 +98,87 @@ def article_exists(obj, eng):
     """
     matches = dedupe_list(match(obj.data))
     record_ids = [el['_source']['control_number'] for el in matches]
-    if record_ids:
-        obj.extra_data['record_matches'] = record_ids
-        return True
+    obj.extra_data.setdefault('matches', {})['exact'] = record_ids
+    return bool(record_ids)
 
-    obj.extra_data['record_matches'] = []
-    return False
+
+@with_debug_logging
+def fuzzy_match(obj, eng):
+    """Return ``True`` if a similar record is found in the system.
+
+    Uses a custom configuration for ``inspire-matcher`` to find records
+    similar to the current workflow object's payload in the system.
+
+    Also sets the ``matches.fuzzy`` property in ``extra_data`` to the list of
+    control numbers that matched.
+
+    Arguments:
+        obj: a workflow object.
+        eng: a workflow engine.
+
+    Returns:
+        bool: ``True`` if the workflow object has a duplicate in the system
+        ``False`` otherwise.
+
+    """
+    fuzzy_match_configuration = {
+        'algorithm': [
+            {
+                'queries': [
+                    {
+                        'clauses': [
+                            {
+                                'boost': 20,
+                                'path':  'abstracts',
+                            },
+                            {
+                                'boost': 10,
```

Comment:
@jacquerie I have updated the configuration tin according to https://github.com/inspirehep/invenio-matcher-benchmark/blob/83caef0aaddb59269cb4769fd1fc9d72fb8234f2/config.py, but removing the `boost` from `authors` I couldn't match the basic record I am using. Shall we go for some benchmarks? If yes, I think they are out of the scope of this PR.